plugin/lighthouse/lighthouse.go

```diff
@@ -1,38 +1,15 @@
 package lighthouse
 
 import (
-	"context"
-
 	"github.com/coredns/coredns/plugin"
 	clog "github.com/coredns/coredns/plugin/pkg/log"
-	"github.com/coredns/coredns/plugin/pkg/nonwriter"
-	"github.com/coredns/coredns/request"
-
-	"github.com/miekg/dns"
 )
 
 // Define log to be a logger with the plugin name in it. This way we can just use log.Info and
 // friends to log.
 var log = clog.NewWithPlugin("lighthouse")
 
 type Lighthouse struct {
-	Next plugin.Handler
-}
-
-// ServeDNS implements the plugin.Handler interface.
-func (lh Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
-	state := request.Request{W: w, Req: r}
-	// Debug log that we've have seen the query. This will only be shown when the debug plugin is loaded.
-	log.Debugf("Received query from '%s'", state.Name())
-
-	// Use a non-writer so we don't write NXDOMAIN to client
-	nw := nonwriter.New(w)
-
-	// Call next plugin (if any).
-	return plugin.NextOrFailure(lh.Name(), lh.Next, ctx, nw, r)
-}
-
-// Name implements the Handler interface.
-func (lh Lighthouse) Name() string {
-	return "lighthouse"
+	Next    plugin.Handler
+	SvcsMap map[string]string
```

Comment:
Just want to add this, as it was not commented and it's something I'd comment in the f-up patch. Since you are using this map as a return type in other places it makes sense to create a type for  it and make it more explicit.

Reply:
This is just placeholder map. Will be replaced eventually by cache+reader for our CRD. this is just to allow some e2e testing of DNS code while CRD cache+reader is done. But good idea, will get more familiar with how to create types.